cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/ProgramLifecycleServiceAuthorizationTest.java

```diff
@@ -0,0 +1,120 @@
+import co.cask.cdap.internal.AppFabricTestHelper;
+import co.cask.cdap.internal.test.AppJarHelper;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramType;
+import co.cask.cdap.proto.id.InstanceId;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.security.Action;
+import co.cask.cdap.proto.security.Principal;
+import co.cask.cdap.security.authorization.AuthorizationEnforcementService;
+import co.cask.cdap.security.authorization.AuthorizerInstantiator;
+import co.cask.cdap.security.authorization.InMemoryAuthorizer;
+import co.cask.cdap.security.spi.authentication.SecurityRequestContext;
+import co.cask.cdap.security.spi.authorization.Authorizer;
+import com.google.inject.Injector;
+import org.apache.twill.filesystem.LocalLocationFactory;
+import org.apache.twill.filesystem.Location;
+import org.apache.twill.filesystem.LocationFactory;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+
+/**
+ * Test authorization for ProgramLifeCycleService
+ */
+public class ProgramLifecycleServiceAuthorizationTest {
+
+  @ClassRule
+  public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
+  private static final Principal ALICE = new Principal("alice", Principal.PrincipalType.USER);
+
+  private static CConfiguration cConf;
+  private static Authorizer authorizer;
+  private static AuthorizationEnforcementService authorizationEnforcementService;
+  private static AppFabricServer appFabricServer;
+  private static ProgramLifecycleService programLifecycleService;
+  private static RemoteSystemOperationsService remoteSystemOperationsService;
+
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    cConf = createCConf();
+    Injector injector = AppFabricTestHelper.getInjector(cConf);
+    authorizer = injector.getInstance(AuthorizerInstantiator.class).get();
+    authorizer.grant(new InstanceId(cConf.get(Constants.INSTANCE_NAME)), ALICE, Collections.singleton(Action.ADMIN));
+    authorizationEnforcementService = injector.getInstance(AuthorizationEnforcementService.class);
+    authorizationEnforcementService.startAndWait();
+    appFabricServer = injector.getInstance(AppFabricServer.class);
+    appFabricServer.startAndWait();
+    programLifecycleService = injector.getInstance(ProgramLifecycleService.class);
+    remoteSystemOperationsService = injector.getInstance(RemoteSystemOperationsService.class);
+    remoteSystemOperationsService.startAndWait();
+  }
+
+  @Test
+  public void testProgramList() throws Exception {
+    SecurityRequestContext.setUserId(ALICE.getName());
+    authorizer.grant(NamespaceId.DEFAULT, ALICE, Collections.singleton(Action.WRITE));
+    AppFabricTestHelper.deployApplication(Id.Namespace.DEFAULT, AllProgramsApp.class, "{}", cConf);
```

Comment:
it seems like config is ``@Nullable``, so you can just pass null, instead of ``"{}"``